### PR TITLE
Create docs for AWS CodePipeline integration

### PR DESCRIPTION
**Pull request by andrea-mosk**
### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+   }
+ }
+ ```
+
+The event pattern above sets up the integration only for the `first-pipeline` and `second-pipeline` pipelines.
+
+### Correlate pipelines with tests
+
+If you are using the [Test Visibility][8] product and your pipeline contains one or more [AWS CodeBuild][9] actions to execute tests, you can correlate your tests
+with the related pipeline inside Datadog's CI Visibility product. To do that:
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:92]: "with the related pipeline inside Datadog Pipeline Visibility."

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+1. In the AWS Console, go to your pipeline configuration and click on **Edit**
+2. Go to the stage containing the AWS CodeBuild action, click on **Edit Stage**, and then edit the relevant action.
+3. Under **Environment variables**, add a new environment variable. The name of the variable should be `DD_PIPELINE_EXECUTION_ID`
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:95]: "Name the variable `DD_PIPELINE_EXECUTION_ID`, and the value `#{codepipeline.PipelineExecutionId}`. Leave the type as _Plaintext_."

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     "source": ["aws.codepipeline"],
+     "detail-type": ["CodePipeline Pipeline Execution State Change", "CodePipeline Action Execution State Change", "CodePipeline Stage Execution State Change"]
+   }
+   ```
+   The JSON above sets up the integration for all of your pipelines. To restrict the set of pipelines,
+   follow the [Only monitor specific pipelines][6] section below.
+6. Click on **Next**.
+7. Under **Target Types**, select **EventBridge API destination**. Then, choose **Use an existing API Destination**
+and select the API destination that you have created in the previous step. Alternatively, you can also create the API destination
+at this moment by following the steps outlined in the [Create the API Destination][7] section.
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:65]: remove "at this moment"

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+8. Under **Headers Parameters**, Click on **Add header parameter**. Input **DD-CI-PROVIDER-AWSCODEPIPELINE** as the key and **true** as the value.
+9. Choose **Create a new role for this specific resource** (or use an existing one if you prefer).
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:67]: remove "if you prefer"

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+10. Review that the information is correct and create the rule.
+
+Once the rule is created, the integration is complete, and you can monitor your pipelines in Datadog.
+
+### Only monitor specific pipelines
+
+You can optionally restrict the pipelines that are monitored by the CI Visibility product.
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:74]: "by Pipeline Visibility"

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/pipelines/_index.md [hunk 6]
```diff
@@ -28,23 +28,24 @@ cascade:
...
-| {{< ci-details title="Pipeline failure reason" >}}Identification of a specific reason behind a pipeline or job failure.{{< /ci-details >}} | {{< X >}} | {{< X >}} |  |  |  |  | {{< X >}} | {{< X >}} |
+|  | Jenkins | GitLab | CircleCI | Buildkite | GitHub Actions | Azure Pipelines | Codefresh | TeamCity | AWS CodePipeline |
+|---|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| {{< ci-details title="Pipeline trace visualization" >}}Visualization of pipeline executions with associated tracing.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
+| {{< ci-details title="Partial retries" >}}Identification of partial retries (for example, when only a subset of jobs were retried).{{< /ci-details >}} |  | {{< X >}} |  | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
```
[the remark this answers is not in the document]
Comment (andrea-mosk) [reply content/en/continuous_integration/pipelines/_index.md:34]: Mmm, from what I see in the guide we should avoid using "e.g." in favour of "for example"

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+The following guide separates the two creation steps for clarity, but note that it's not required to create the resources
+separately. They can also be created at the same time during the EventBridge Rule creation process.
+
+More information about monitoring pipeline events can be found in the [official AWS guide][4].
+
+### Create the API Destination
+
+1. In the AWS Console, go to **EventBridge > API destinations** and click on **Create API destination**.
+2. Choose a name for the API Destination (for example, **datadog-ci-visibility-api**) and optionally add a description.
```
[the remark this answers is not in the document]
Comment (andrea-mosk) [reply content/en/continuous_integration/pipelines/awscodepipeline.md:37]: (see comment above)

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+3. Under **API destination endpoint**, input <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook</code>
+4. Under **HTTP method**, select **POST**.
+5. Under Connection type, select **Create a new connection**:
+   1. Choose a name for the connection (for example, **datadog-ci-visibility-connection**) and optionally add a description.
```
[the remark this answers is not in the document]
Comment (andrea-mosk) [reply content/en/continuous_integration/pipelines/awscodepipeline.md:41]: (see comment above)

File: content/en/continuous_integration/pipelines/_index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (kassenq) [reply content/en/continuous_integration/pipelines/_index.md:34]: Wow I totally read that the other way around. Please ignore this then 😂

**Review by kassenq — APPROVED** [review]

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+## Compatibility
+
+- **Partial pipelines**: View partially retried executions
+
+## Configure the Datadog integration
+
+To set up the integration between [AWS CodePipeline][1] and Datadog CI Visibility, two AWS resources need to be created:
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:25]:
```suggestion
To set up the integration between [AWS CodePipeline][1] and Datadog CI Visibility, create two AWS resources  during the EventBridge Rule creation process:
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+1. [API Destination][2]: an HTTP endpoint pointing to Datadog's intake.
+2. [AWS EventBridge Rule][3]: a rule that forwards CodePipeline events to the API Destination.
+
+The following guide separates the two creation steps for clarity, but note that it's not required to create the resources
+separately. They can also be created at the same time during the EventBridge Rule creation process.
+
+More information about monitoring pipeline events can be found in the [official AWS guide][4].
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:32]:
```suggestion
You can create these resources separately, or at the same time. For more information about monitoring pipeline events, see the [official AWS guide][4].
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+### Create the API Destination
+
+1. In the AWS Console, go to **EventBridge > API destinations** and click on **Create API destination**.
+2. Choose a name for the API Destination (for example, **datadog-ci-visibility-api**) and optionally add a description.
+3. Under **API destination endpoint**, input <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook</code>
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:38]:
```suggestion
3. Under **API destination endpoint**, input <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook</code>.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+4. Under **HTTP method**, select **POST**.
+5. Under Connection type, select **Create a new connection**:
+   1. Choose a name for the connection (for example, **datadog-ci-visibility-connection**) and optionally add a description.
+   2. Under **Destination type**, select **Other**
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:42]:
```suggestion
   2. Under **Destination type**, select **Other**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+   3. Under **Authorization type**, select **API key**.
+   Input **DD-API-KEY** as the **API key name** and input your [Datadog API Key][5] in the **Value** field.
+6. Click on **Create**.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:45]:
```suggestion
6. Click **Create**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+2. [AWS EventBridge Rule][3]: a rule that forwards CodePipeline events to the API Destination.
+
+The following guide separates the two creation steps for clarity, but note that it's not required to create the resources
+separately. They can also be created at the same time during the EventBridge Rule creation process.
+
+More information about monitoring pipeline events can be found in the [official AWS guide][4].
+
+### Create the API Destination
+
+1. In the AWS Console, go to **EventBridge > API destinations** and click on **Create API destination**.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:36]:
```suggestion
1. In the AWS Console, go to **EventBridge > API destinations** and click **Create API destination**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+5. Under Connection type, select **Create a new connection**:
+   1. Choose a name for the connection (for example, **datadog-ci-visibility-connection**) and optionally add a description.
+   2. Under **Destination type**, select **Other**
+   3. Under **Authorization type**, select **API key**.
+   Input **DD-API-KEY** as the **API key name** and input your [Datadog API Key][5] in the **Value** field.
+6. Click on **Create**.
+
+### Create the EventBridge Rule
+
+1. In the AWS Console, go to **EventBridge > Rules** and click on **Create Rule**.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:49]:
```suggestion
1. In the AWS Console, go to **EventBridge > Rules** and click **Create Rule**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+2. Choose a name for the rule (for example, **datadog-ci-visibility-integration**) and optionally add a description.
+3. Leave the event bus as **default**, and, under **Rule Type**, select **Rule with an event pattern**. Click on **Next**.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:51]:
```suggestion
3. Leave the event bus as **default**, and, under **Rule Type**, select **Rule with an event pattern**. Click **Next**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+5. Under **Creation Method**, select **Custom pattern (JSON editor)**. Then, under **Event Pattern**, input the following JSON:
+   ```
+   {
+     "source": ["aws.codepipeline"],
+     "detail-type": ["CodePipeline Pipeline Execution State Change", "CodePipeline Action Execution State Change", "CodePipeline Stage Execution State Change"]
+   }
+   ```
+   The JSON above sets up the integration for all of your pipelines. To restrict the set of pipelines,
+   follow the [Only monitor specific pipelines][6] section below.
+6. Click on **Next**.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:62]:
```suggestion
6. Click **Next**.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+7. Under **Target Types**, select **EventBridge API destination**. Then, choose **Use an existing API Destination**
+and select the API destination that you have created in the previous step. Alternatively, you can also create the API destination
+by following the steps outlined in the [Create the API Destination][7] section.
+8. Under **Headers Parameters**, Click on **Add header parameter**. Input `DD-CI-PROVIDER-AWSCODEPIPELINE` as the key and `true` as the value.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:66]:
```suggestion
8. Under **Headers Parameters**, click **Add header parameter**. Input `DD-CI-PROVIDER-AWSCODEPIPELINE` as the key and `true` as the value.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+   }
+ }
+ ```
+
+The event pattern above sets up the integration only for the `first-pipeline` and `second-pipeline` pipelines.
+
+### Correlate pipelines with tests
+
+If you are using [Test Visibility][8] and your pipeline contains one or more [AWS CodeBuild][9] actions to execute tests, you can correlate your tests
+with the related pipeline inside Datadog Pipeline Visibility. To do that:
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:92]:
```suggestion
with the related pipeline inside Datadog Pipeline Visibility. 

```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+1. In the AWS Console, go to your pipeline configuration and click on **Edit**
+2. Go to the stage containing the AWS CodeBuild action, click on **Edit Stage**, and then edit the relevant action.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:94]:
```suggestion
1. In the AWS Console, go to your pipeline configuration and click **Edit**
2. Go to the stage containing the AWS CodeBuild action, click **Edit Stage**, and then edit the relevant action.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+3. Under **Environment variables**, add a new environment variable.
+Name the variable `DD_PIPELINE_EXECUTION_ID`, and the value `#{codepipeline.PipelineExecutionId}`. Leave the type as _Plaintext_.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:96]:
```suggestion
3. Under **Environment variables**, add an environment variable.
Name the variable `DD_PIPELINE_EXECUTION_ID`, and the value `#{codepipeline.PipelineExecutionId}`. Leave the type as _Plaintext_.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+4. Click on **Done** to save your changes.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:97]:
```suggestion
4. Click **Done** to save your changes.
```

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+
+The steps above allow you to add the pipeline execution ID to your CodeBuild action environment variables. More information
+can be found in the official AWS guide on [Working with Variables][10].
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:100]:
```suggestion
The steps above allow you to add the pipeline execution ID to your CodeBuild action environment variables. For more information on working with variables, see the [official AWS guide][10].
```

**Review by alai97 — COMMENTED** [review]
Some copy suggestions!

File: content/en/continuous_integration/pipelines/awscodepipeline.md (hunk 9 above)
Comment (andrea-mosk) [reply content/en/continuous_integration/pipelines/awscodepipeline.md:25]: Mmm I would prefer to not have `during the EventBridge Rule creation process` there, as also in the guide they are created separately. So we could go for an hybrid like:

```suggestion
To set up the integration between [AWS CodePipeline][1] and Datadog CI Visibility, create two AWS resources:
```

?

File: content/en/continuous_integration/pipelines/awscodepipeline.md (hunk 10 above)
Comment (andrea-mosk) [reply content/en/continuous_integration/pipelines/awscodepipeline.md:32]: And here perhaps we can re-add the "during the EventBridge Rule creation process" from above:

```suggestion
You can create these resources separately, or at the same time, during the EventBridge Rule creation process. For more information about monitoring pipeline events, see the [official AWS guide][4].
```

what do you think?

**Review by alai97 — APPROVED** [review]
Looks good for docs!

File: content/en/continuous_integration/pipelines/awscodepipeline.md (hunk 10 above)
Comment (kassenq) [reply content/en/continuous_integration/pipelines/awscodepipeline.md]: LGTM

**Pull request merged**